component/encryptor_v2/encryptor_v2.go

```diff
@@ -0,0 +1,468 @@
+package encryptor_v2
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/Azure/azure-storage-fuse/v2/common/config"
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
+	"github.com/Azure/azure-storage-fuse/v2/internal"
+	"github.com/Azure/azure-storage-fuse/v2/internal/handlemap"
+)
+
+type Encryptor_v2 struct {
+	internal.BaseComponent
+	dataFileHandle   *os.File
+	metaFileHandle   *os.File
+	blockSize        uint64
+	mountPointCipher string
+	encryptionKey    []byte
+	tmpPath          string
+}
+
+type LastChunkMeta struct {
+	sync.Mutex
+	farthestBlockSeen int64 // keep track of the farthest block that has been written to the file
+	paddingLength     int64 // keep track of the padding length of the last block
+}
+
+var lastchunkMeta = &LastChunkMeta{
+	farthestBlockSeen: -1,
+	paddingLength:     0,
+}
+
+type EncryptorOptions struct {
+	BlockSize          uint64 `config:"block-size-mb" yaml:"block-size-mb,omitempty"`
+	EncryptedMountPath string `config:"encrypted-mount-path" yaml:"encrypted-mount-path,omitempty"`
+	EncryptionKey      string `config:"encryption-key" yaml:"encryption-key,omitempty"`
+	TmpPath            string `config:"path" yaml:"path,omitempty"`
+}
+
+const (
+	compName    = "encryptor_v2"
```

Comment:
>encryptor_v2 [](http://example.com/codeflow?start=16&length=12)

is the plan to keep both around for the long term or we will make the call to remove one soon?

Reply:
I am doing some performance comparison after that will remove one.